narwhals/_pandas_like/expr.py

```diff
@@ -345,3 +357,117 @@ def total_microseconds(self) -> PandasExpr:
 
     def total_nanoseconds(self) -> PandasExpr:
         return register_namespace_expression_call(self._expr, "dt", "total_nanoseconds")
+
+
+class PandasExprNameNamespace:
+    def __init__(self, expr: PandasExpr) -> None:
+        self._expr = expr
+
+    def map(self, function: Callable[[str], str]) -> PandasExpr:
+        root_names = self._expr._root_names
+
+        if root_names is None:
+            msg = (
+                "Anonymous expressions are not supported in `.name.map`.\n"
+                "Instead of `nw.all()`, try using a named expression, such as "
+                "`nw.col('a', 'b')`\n"
+            )
+            raise ValueError(msg)
+
+        return self._expr.__class__(
+            lambda df: [
+                series.alias(function(series.name)) for series in self._expr._call(df)
+            ],
+            depth=self._expr._depth,
+            function_name=self._expr._function_name,
+            root_names=root_names,
+            output_names=[function(name) for name in root_names],
+            implementation=self._expr._implementation,
+        )
+
+    def prefix(self, prefix: str) -> PandasExpr:
+        root_names = self._expr._root_names
+        if root_names is None:
+            msg = (
+                "Anonymous expressions are not supported in `.name.prefix`.\n"
+                "Instead of `nw.all()`, try using a named expression, such as "
+                "`nw.col('a', 'b')`\n"
+            )
+            raise ValueError(msg)
+
+        return self._expr.__class__(
+            lambda df: [
+                series.alias(prefix + series.name) for series in self._expr._call(df)
+            ],
+            depth=self._expr._depth,
+            function_name=self._expr._function_name,
+            root_names=root_names,
+            output_names=[prefix + name for name in root_names],
+            implementation=self._expr._implementation,
+        )
+
+    def suffix(self, suffix: str) -> PandasExpr:
+        root_names = self._expr._root_names
+
+        if root_names is None:
+            msg = (
+                "Anonymous expressions are not supported in `.name.suffix`.\n"
+                "Instead of `nw.all()`, try using a named expression, such as "
+                "`nw.col('a', 'b')`\n"
+            )
+            raise ValueError(msg)
+
+        return self._expr.__class__(
+            lambda df: [
+                series.alias(series.name + suffix) for series in self._expr._call(df)
+            ],
+            depth=self._expr._depth,
+            function_name=self._expr._function_name,
+            root_names=root_names,
+            output_names=[name + suffix for name in root_names],
```

Comment:
Sometimes github loses notification I guess! Anyway, due to 

> This will undo any previous renaming operations on the expression.

I thought applying to root_names was the correct way of doing it. But from the comment below I understand it is not enough :(

Reply:
you might be able to do it on the root names, and then apply the transformations to
```
            lambda df: [
                series.alias(function(series.name)) for series in self._expr._call(df)
            ],
```

but the number of output names might not match the number of root names. I'm a little worried about introducing bugs here, we may need to study the Polars code for this operation first (I've not looked at that part of the codebase yet)